src/Configuration/Types.hs

```diff
@@ -0,0 +1,383 @@
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+
+module Configuration.Types
+    ( Cluster(..)
+    , ClusterConfig(..)
+    , Host(..)
+    , InstallerConfig (..)
+    , Launcher(..)
+    , LauncherConfig(..)
+    , NodeArgs(..)
+    , OS(..)
+    , OSConfig(..)
+    , Pass(..)
+    , TopologyConfig(..)
+    , WalletConfig(..)
+    , renderOS
+    , renderCluster
+    ) where
+
+import           Cardano.Prelude hiding (evalState)
+
+import           Control.Monad.Trans.State.Strict (evalState)
+import           Data.Functor.Contravariant (contramap)
+import qualified Data.Text as T
+import           Dhall (GenericInject, GenericInterpret, Inject (..), InputType,
+                        Interpret (..), InterpretOptions (..), Type, auto,
+                        field, genericAutoWith, genericInjectWith, record,
+                        strictText)
+import           GHC.Generics (from, to)
+import           Test.QuickCheck (Arbitrary (..), Gen, arbitraryASCIIChar,
+                                  choose, elements, listOf, listOf1)
+
+-- | Operating system
+data OS
+    = Linux64
+    | Macos64
+    | Win64
+    deriving (Bounded, Enum, Eq, Read, Show)
```

Comment:
Maybe you don't need `Show` if you have specific functions for rendering these into `Text`. Do you really need anything more than `Eq` here?